### PR TITLE
Paint Mini Golf

### PR DESCRIPTION
**Pull request by marijnvdwerf**
image

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
@duncanspumpkin This one is really different compared to the others I've done.

**Comment by marijnvdwerf** [comment]
Found out what `0x141E9B4 - 0x141E9D4` mean. They define whether metal supports can go through tile segments.

image

**Comment by duncanspumpkin** [comment]
@marijnvdwerf ah that makes sense. We could do with maybe making some easier technique for setting those variables then. Maybe a function that takes heights for each location:

```
void set_current_tile_support_heights( int x0y0, int x1y0, int x2y0, ...){
    RCT2_GLOBAL(0x141E9B4, sint16) = x0y0;
    ...
}
```
What do you think?

**Comment by marijnvdwerf** [comment]
Personally I've been thinking about returning it from the tile drawing
function. As it's kinda a side-effect of drawing the sprite. I guess we
need a struct for it either way, and I'd prefer that to have
keys/properties like 'b4' etc, because that would be easier to work with
while decompiling, as the order differs
Op vr 8 apr. 2016 om 13:58 schreef Duncan <notifications@github.com>

**Comment by marijnvdwerf** [comment]
Also, not all paint functions update all segment heights. For example, golf course pathways only block 3 or 4 segments. 
image

File: src/interface/viewport.c [hunk 1]
```diff
@@ -1113,6 +1113,72 @@ int sub_98198C(int image_id, sint8 x_offset, sint8 y_offset, sint16 length_x, si
...
+	uint32 eax = RCT2_ADDRESS(0x008B8F74, uint32)[vehicle->var_D4];
+	uint32 ebx = (RCT2_GLOBAL(eax + vehicle->var_C5, uint8) << 2) + (imageDirection >> 3);
+
+	uint32 image_id = rideType->vehicles[0].base_image_id + 1 + ebx;
+
+	uint32 peep_palette = sprite->peep.tshirt_colour << 19 | sprite->peep.trousers_colour << 24 | 0x0A0000000;
+
+	RCT2_GLOBAL(0x9DEA52, uint16) = 0;
+	RCT2_GLOBAL(0x9DEA54, uint16) = 0;
+	RCT2_GLOBAL(0x9DEA56, uint16) = z + 5;
```
Comment (zsilencer) [inline src/interface/viewport.c]: these are used as arguments to sub_98197C so they are not needed, and you forgot to add 5 to z on that, same with ball.

File: src/ride/gentle/mini_golf.c [hunk 2]
```diff
@@ -13,3 +13,1007 @@
...
+	}
+
+	rct_xy16 boundBox = (supportType & 1) ? (rct_xy16) {26, 32} : (rct_xy16) {32, 26};
+	rct_xy16 boundBoxOffset = (supportType & 1) ? (rct_xy16) {3, 0} : (rct_xy16) {0, 3};
+
+	imageId = mini_golf_track_sprites_hole_e[direction][trackSequence][1] | RCT2_GLOBAL(0x00F44198, uint32);
+
+	switch ((direction << 4) | trackSequence) {
+		case 0x01:
+			sub_98197C(imageId, 0, 0, 2, 26, 3, height, 30, 3, height + 4, get_current_rotation());
```
Comment (janisozaur) [inline src/ride/gentle/mini_golf.c]: no `break;`?

**Pull request merged**